sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -509,6 +509,7 @@ private[sql] class JDBCRDD(
       } catch {
         case e: Exception => logWarning("Exception closing connection", e)
       }
+      closed = true
```

Comment:
I think this should go at the top, right after it is checked. My reasoning is that if close fails, it would let itself be called again, and that's not obviously helpful, I think.